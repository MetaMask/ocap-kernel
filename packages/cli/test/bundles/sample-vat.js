/**
 * Start function for generic test vat.
 *
 * @param {unknown} parameters - Initialization parameters from the vat's config object.
 * @returns {unknown} The root object for the new vat.
 */
export function start(parameters) {
  const name = parameters?.name ?? 'anonymous';
  console.log(`start vat root object "${name}"`);
  return {
    name,
    stuff: `initialized with ${JSON.stringify(parameters)}`,
  };
}
