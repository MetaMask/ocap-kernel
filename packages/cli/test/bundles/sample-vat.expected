{"moduleFormat":"endoZipBase64","endoZipBase64":"UEsDBAoAAAAAAAAAAADY08YWYwIAAGMCAAAUAAAAY29tcGFydG1lbnQtbWFwLmpzb257CiAgInRhZ3MiOiBbXSwKICAiZW50cnkiOiB7CiAgICAiY29tcGFydG1lbnQiOiAiQG9jYXAvY2xpLXYwLjAuMCIsCiAgICAibW9kdWxlIjogIi4vdGVzdC9zdGFnZS9idW5kbGVzL3NhbXBsZS12YXQuanMiCiAgfSwKICAiY29tcGFydG1lbnRzIjogewogICAgIkBvY2FwL2NsaS12MC4wLjAiOiB7CiAgICAgICJuYW1lIjogIkBvY2FwL2NsaSIsCiAgICAgICJsYWJlbCI6ICJAb2NhcC9jbGktdjAuMC4wIiwKICAgICAgImxvY2F0aW9uIjogIkBvY2FwL2NsaS12MC4wLjAiLAogICAgICAibW9kdWxlcyI6IHsKICAgICAgICAiLi90ZXN0L3N0YWdlL2J1bmRsZXMvc2FtcGxlLXZhdC5qcyI6IHsKICAgICAgICAgICJsb2NhdGlvbiI6ICJ0ZXN0L3N0YWdlL2J1bmRsZXMvc2FtcGxlLXZhdC5qcyIsCiAgICAgICAgICAicGFyc2VyIjogInByZS1tanMtanNvbiIsCiAgICAgICAgICAic2hhNTEyIjogIjA0ZmI2Y2EwMTk0Mzc2ZjIwMzE5ZGEwMjQwYWM5OTg0NDM2MTQ5MTU0MDgzZjQ4ZDIzMWFjNTI5ZTEyZmNlZGRjYmU0OTZlNzViYTg2ZGFjYTcyY2Q5OTVlZDNkNzQ5NWY5NmNmODM1OWY1ZGRiYjk2MmM4YmY3ZTMyYjg3YWYyIgogICAgICAgIH0KICAgICAgfQogICAgfQogIH0KfVBLAwQKAAAAAAAAAAAAPl/YgU0DAABNAwAAMQAAAEBvY2FwL2NsaS12MC4wLjAvdGVzdC9zdGFnZS9idW5kbGVzL3NhbXBsZS12YXQuanN7ImltcG9ydHMiOltdLCJleHBvcnRzIjpbInN0YXJ0Il0sInJlZXhwb3J0cyI6W10sIl9fc3luY01vZHVsZVByb2dyYW1fXyI6Iih7ICAgaW1wb3J0czogJGjNj19pbXBvcnRzLCAgIGxpdmVWYXI6ICRozY9fbGl2ZSwgICBvbmNlVmFyOiAkaM2PX29uY2UsICAgaW1wb3J0TWV0YTogJGjNj19fX19tZXRhLCB9KSA9PiAoZnVuY3Rpb24gKCkgeyAndXNlIHN0cmljdCc7ICAgJGjNj19pbXBvcnRzKFtdKTtPYmplY3QuZGVmaW5lUHJvcGVydHkoc3RhcnQsICduYW1lJywge3ZhbHVlOiBcInN0YXJ0XCJ9KTskaM2PX29uY2Uuc3RhcnQoc3RhcnQpOyAgIC8qKlxuICogU3RhcnQgZnVuY3Rpb24gZm9yIGdlbmVyaWMgdGVzdCB2YXQuXG4gKlxuICogQHBhcmFtIHt1bmtub3dufSBwYXJhbWV0ZXJzIC0gSW5pdGlhbGl6YXRpb24gcGFyYW1ldGVycyBmcm9tIHRoZSB2YXQncyBjb25maWcgb2JqZWN0LlxuICogQHJldHVybnMge3Vua25vd259IFRoZSByb290IG9iamVjdCBmb3IgdGhlIG5ldyB2YXQuXG4gKi9cbmZ1bmN0aW9uICAgICAgICBzdGFydChwYXJhbWV0ZXJzKXtcbmNvbnN0IG5hbWU9cGFyYW1ldGVycz8ubmFtZT8/J2Fub255bW91cyc7XG5jb25zb2xlLmxvZyggYHN0YXJ0IHZhdCByb290IG9iamVjdCBcIiR7bmFtZX1cImApO1xucmV0dXJue1xubmFtZSxcbnN0dWZmOiBgaW5pdGlhbGl6ZWQgd2l0aCAke0pTT04uc3RyaW5naWZ5KHBhcmFtZXRlcnMpfWB9O1xuXG4gfVxufSkoKVxuIiwiX19saXZlRXhwb3J0TWFwX18iOnt9LCJfX3JlZXhwb3J0TWFwX18iOnt9LCJfX2ZpeGVkRXhwb3J0TWFwX18iOnsic3RhcnQiOlsic3RhcnQiXX0sIl9fbmVlZHNJbXBvcnRNZXRhX18iOmZhbHNlfVBLAQIeAwoAAAAAAAAAAADY08YWYwIAAGMCAAAUAAAAAAAAAAAAAACkgQAAAABjb21wYXJ0bWVudC1tYXAuanNvblBLAQIeAwoAAAAAAAAAAAA+X9iBTQMAAE0DAAAxAAAAAAAAAAAAAACkgZUCAABAb2NhcC9jbGktdjAuMC4wL3Rlc3Qvc3RhZ2UvYnVuZGxlcy9zYW1wbGUtdmF0LmpzUEsFBgAAAAACAAIAoQAAADEGAAAAAA==","endoZipBase64Sha512":"bcefe2bfbe1f9730d46eccd9f93bcca779f223e2d0a79dbc25bac6b86478a66435bf2298cf0c3c4dc6fd5728e38962a534c66c45f21469cfaafd5c31e967713a"}